lib/index.ts

```diff
@@ -769,7 +769,7 @@ export class Server<
    *
    * @public
    */
-  public fetchSockets(): Promise<RemoteSocket<EmitEvents>[]> {
```

Comment:
So we would have:

```js
const sockets = await io.fetchSockets();

// with the BroadcastOperator
const sockets = await io.in("room-1").fetchSockets<SocketData>();
```

Is that right?

Reply:
Exactly

Reply:
In that case, I think it would make more sense to go with your first solution [here](https://github.com/socketio/socket.io/issues/4229#issuecomment-1004182988), so we have the same API with and without operator (like `to()` or `except()`).

```ts
const sockets = await io.fetchSockets();
const sockets = await io.in("room-1").fetchSockets(); // no additional types
```

What do you think?

Reply:
Sounds good to me. Will update the PR